include/ddc/kernels/splines/splines_linear_problem_band.hpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cassert>
+#include <memory>
+#include <string>
+
+#if __has_include(<mkl_lapacke.h>)
+#include <mkl_lapacke.h>
+#else
+#include <lapacke.h>
+#endif
+
+#include "splines_linear_problem.hpp"
+
+namespace ddc::detail {
+
+/**
+ * @brief A band linear problem dedicated to the computation of a spline approximation.
+ *
+ * The storage format is dense row-major. Lapack is used to perform every matrix and linear solver-related operations.
+ *
+ * @tparam ExecSpace The Kokkos::ExecutionSpace on which operations related to the matrix are supposed to be performed. Note: atm this is a placeholder for futur developments on GPU.
+ */
+template <class ExecSpace>
+class SplinesLinearProblemBand : public SplinesLinearProblem<ExecSpace>
+{
+public:
+    using typename SplinesLinearProblem<ExecSpace>::MultiRHS;
+    using SplinesLinearProblem<ExecSpace>::size;
+
+protected:
+    const std::size_t m_kl; // no. of subdiagonals
+    const std::size_t m_ku; // no. of superdiagonals
+    const std::size_t m_c; // no. of "rows" in q (in the sense of a band format storage)
```

Comment:
```suggestion
    std::size_t m_kl; // no. of subdiagonals
    std::size_t m_ku; // no. of superdiagonals
    std::size_t m_c; // no. of "rows" in q (in the sense of a band format storage)
```

Reply:
Can you explain me why we do not need const here ?